lib/swm050/gpio.c

```diff
@@ -0,0 +1,173 @@
+ * it under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this library.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include <libopencm3/swm050/gpio.h>
+
+/*---------------------------------------------------------------------------*/
+/** @brief Set a Group of Pins
+
+Set one or more pins of GPIO to 1. Please note that this chip doesn't support
+atomic pin setting.
+
+@param[in] gpios Unsigned int16. Pin identifiers @ref gpio_pin_id
+	     If multiple pins are to be changed, use bitwise OR '|' to separate
+	     them.
+*/
+void gpio_set(uint16_t gpios)
+{
+	GPIO_DATA |= gpios;
+}
+
+/*---------------------------------------------------------------------------*/
+/** @brief Clear a Group of Pins
+
+Set one or more pins of GPIO to 0. Please note that this chip doesn't support
+atomic pin setting.
+
+@param[in] gpios Unsigned int16. Pin identifiers @ref gpio_pin_id
+	     If multiple pins are to be changed, use bitwise OR '|' to separate
+	     them.
+*/
+void gpio_clear(uint16_t gpios)
+{
+	GPIO_DATA &= ~gpios;
+}
+
+/*---------------------------------------------------------------------------*/
+/** @brief Read a Group of Pins.
+
+@param[in] gpios Unsigned int16. Pin identifiers @ref gpio_pin_id
+	    If multiple pins are to be read, use bitwise OR '|' to separate
+	    them.
+@return Unsigned int16 value of the pin values. The bit position of the pin
+	value returned corresponds to the pin number.
+*/
+uint16_t gpio_get(uint16_t gpios)
+{
+	return GPIO_EXT & gpios;
+}
+
+/*---------------------------------------------------------------------------*/
+/** @brief Toggle a Group of Pins
+
+Toggle one or more pins of GPIO. The non-toggled pins are not affected.
+
+@param[in] gpios Unsigned int16. Pin identifiers @ref gpio_pin_id
+	     If multiple pins are to be changed, use bitwise OR '|' to separate
+	     them.
+*/
+void gpio_toggle(uint16_t gpios)
+{
+	uint32_t curr_status = GPIO_DATA & gpios;
+	GPIO_DATA = (GPIO_DATA & (~gpios)) | (~curr_status);
+}
+
+/*---------------------------------------------------------------------------*/
+/** @brief Set the direction of a Group of Pins
+
+Set the direction of one or more pins of GPIO.
+
+@param[in] gpios Unsigned int16. Pin identifiers @ref gpio_pin_id
+	     If multiple pins are to be changed, use bitwise OR '|' to separate
+	     them.
+@param[in] dir Unsigned int8. Pin direction @ref gpio_dir
+*/
+void gpio_dir(uint16_t gpios, uint8_t dir)
+{
+	if (dir == GPIO_INPUT)
```

Comment:
{} here please.  there's a make checkstyle target that should help a bit

Reply:
Oh sorry, I forgot that libopencm3 coding style is a little different than Linux one.